### PR TITLE
[asr->fortran] implement more visitors & tests

### PR DESCRIPTION
**Pull request by khushi-411**

### CODE REVIEW

**Comment by khushi-411** [comment]
I'll add more visitors to this PR.

File: integration_tests/logical1.f90 [hunk 1]
```diff
@@ -1,14 +1,14 @@
 program logical1
-    ! this program checks logical operators
-    implicit none
+! this program checks logical operators
+implicit none
     
-       ! variable declaration
-       logical :: a, b
```
Comment (khushi-411) [inline integration_tests/logical1.f90:6]: The indentations were different from other test files. Hence, I thought to modify it.

File: integration_tests/complex_15.f90 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+program complex_15
+   implicit none
+   complex(8) :: k, z
+   k = (3, -4)
+   z = -k
+   print *, k, z
+
+   if (abs(z) == k) error stop
```
Comment (certik) [inline integration_tests/complex_15.f90:8]: Just indent it to the left. But if you want to indent, use 4 spaces.

Also, let's add a test for what it is supposed to be, something like:
```fortran
if (z /= (-3, 4)) error stop
```

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 3]
```diff
@@ -146,7 +146,7 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
                 r += std::to_string(down_cast<ASR::Real_t>(t)->m_kind);
                 r += ")";
                 break;
-	    } case ASR::ttypeType::Complex: {
+           } case ASR::ttypeType::Complex: {
                 r = "complex(";
                 r += std::to_string(down_cast<ASR::Complex_t>(t)->m_kind);
```
Comment (certik) [inline src/libasr/codegen/asr_to_fortran.cpp:151]: Something is wrong with this indentation, as it should be 4 spaces.

File: integration_tests/complex_15.f90 (hunk 2 above)
Comment (khushi-411) [reply integration_tests/complex_15.f90]: Updated. Thanks, @certik!

File: integration_tests/CMakeLists.txt [hunk 4]
```diff
@@ -1048,6 +1048,7 @@ RUN(NAME complex_11 LABELS gfortran llvmImplicit wasm)
 RUN(NAME complex_12 LABELS gfortran llvm wasm)
 RUN(NAME complex_13 LABELS gfortran llvm wasm)
 RUN(NAME complex_14 LABELS gfortran llvm wasm)
+RUN(NAME complex_15 LABELS gfortran llvm wasm fortran)
```
Comment (certik) [inline integration_tests/CMakeLists.txt:1051]: The CI failed for WASM, so I would remove it:
```suggestion
RUN(NAME complex_15 LABELS gfortran llvm fortran)
```

**Comment by certik** [comment]
The Third Party CI failure is caused by https://github.com/lfortran/lfortran/issues/2776.

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 5]
```diff
@@ -1030,9 +1014,28 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
...
-    // void visit_ComplexUnaryMinus(const ASR::ComplexUnaryMinus_t &x) {}
+    void visit_ComplexUnaryMinus(const ASR::ComplexUnaryMinus_t &x) {
+        visit_expr_with_precedence(*x.m_arg, 9);
+        s = "-" + s;
+        last_expr_precedence = Precedence::UnaryMinus;
+    }
 
-    // void visit_ComplexCompare(const ASR::ComplexCompare_t &x) {}
+    template <typename T>
+    void handle_compare(const T& x) {
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/codegen/asr_to_fortran.cpp:1024]: Move this function to the Utils section above.
Above Line 206

**Comment by khushi-411** [comment]
Thanks, @certik and @Thirumalai-Shaktivel, for reviewing the PR. The CI is green. I think we could merge this if you think it's good to go now.

**Review by certik — APPROVED** [review]
I think this is fine, thanks!

**Pull request merged**